api/samlacs.go

```diff
@@ -148,7 +148,22 @@ func (a *API) SAMLACS(w http.ResponseWriter, r *http.Request) error {
 		}
 	}
 
-	// TODO: fetch new metadata if possible when validUntil < time.Now() or cacheDuration
+	var samlMetadataModified bool
+	samlMetadataModified = false
+	if (!idpMetadata.ValidUntil.IsZero() && idpMetadata.ValidUntil.Before(time.Now())) || ssoProvider.SAMLProvider.UpdatedAt.Add(idpMetadata.CacheDuration).Before(time.Now()) {
```

Comment:
Can you move this condition into a separate function of the `idpMetadata` and `ssoProvider` to increase redability?